tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/JsonDumper.hpp

```diff
@@ -0,0 +1,313 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+#include "AstDumper.hpp"
+#include <iostream>
+#include <nlohmann/json.hpp>
+#include <string>
+#include <string_view>
+
+using namespace nlohmann::literals;
+
+class JsonDumper : public AstDumper {
+  nlohmann::json m_json;
+
+  enum class TokenKinds
+  {
+    Text = 0,
+    Newline = 1,
+    Whitespace = 2,
+    Punctuation = 3,
+    Keyword = 4,
+    LineIdMarker = 5, // use this if there are no visible tokens with ID on the line but you still
+                      // want to be able to leave a comment for it
+    TypeName = 6,
+    MemberName = 7,
+    StringLiteral = 8,
+    Literal = 9,
+    Comment = 10,
+    DocumentRangeStart = 11,
+    DocumentRangeEnd = 12,
+    DeprecatedRangeStart = 13,
+    DeprecatedRangeEnd = 14,
+    SkipDiffRangeStart = 15,
+    SkipDiffRangeEnd = 16,
+    InheritanceInfoStart = 17,
+    InheritanceInfoEnd = 18
+  };
+
+public:
+  JsonDumper(
+      std::string_view reviewName,
+      std::string_view serviceName,
+      std::string_view packageName,
+      std::string_view packageVersion = "")
+      : AstDumper()
+  {
+    m_json["Name"] = reviewName;
+    m_json["Language"] = "C++";
+    m_json["ServiceName"] = serviceName;
+    m_json["PackageName"] = packageName;
+    if (!packageVersion.empty())
```

Comment:
Should this be required?  Seems like it will create problems later if APIView doesn't know the version of the package

Reply:
ApiView doesn't treat it as required.

The packageVersion comes from the command line because we don't really have a way of putting it in the config file - but we do have scripts to extract it from the headers which we can use when it's in the CI pipeline.

There's a bit of sloppiness here because we don't have the CI story fully fleshed out - specifically the steps run during the Azure::Core pipelines aren't fully understood - that's going to need collaboration between Praveen and us. Much of this is a stopgap to get things working until the full CI story is complete.

Reply:
THis means that some of the settings for things like "Review Name" are likely to change.